### ISSUE TITLE
AttributeError: 'Processed' object has no attribute 'sampler'

### ISSUE THREAD

**Issue by cdobrich** [h43lb1t0/BatchCheckpointPrompt#37]
I'm trying to run about 90 different checkpoints through the batch processor. I have tested with Automatic1111 webui Version: v1.3.1 and Version: v1.4.0. Same results. It will process whatever checkpoint I have selected and then throw this error, not processing a second one.  It does not matter if I have ‘Hires. fix' enabled or not. Same error. 

On the front end I'm getting this error message:
```
AttributeError: 'Processed' object has no attribute 'sampler'
Time taken: 52.84sTorch active/reserved: 8170/12570 MiB, Sys VRAM: 17339/24215 MiB (71.6%)
```

On the backend this is the output.

```
################################################################
Install script for stable-diffusion + Web UI
Tested on Debian 11 (Bullseye)
################################################################

################################################################
Running on bob user
################################################################

################################################################
Repo already cloned, using it as install directory
################################################################

################################################################
Create and activate python venv
################################################################

################################################################
Launching launch.py...
################################################################
Using TCMalloc: libtcmalloc.so.4
Python 3.11.2 (main, May 30 2023, 17:45:26) [GCC 12.2.0]
Version: v1.3.1
Commit hash: b6af0a3809ea869fb180633f9affcae4b199ffcf
Installing requirements

Launching Web UI with arguments: --opt-sdp-attention --xformers --listen --no-half-vae --no-half --disable-nan-check
Civitai Helper: Get Custom Model Folder
Civitai Helper: Load setting from: /home/bob/code/repos/stable-diffusion/stable-diffusion-webui/extensions/Stable-Diffusion-Webui-Civitai-Helper/setting.json
Civitai Helper: No setting file, use default
[-] ADetailer initialized. version: 23.6.3, num models: 8
Loading weights [feadfe3cfe] from /home/bob/code/repos/stable-diffusion/stable-diffusion-webui/models/Stable-diffusion/3moonNIReal_3moonNIRealV2.safetensors
Running on local URL:  http://0.0.0.0:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 4.0s (import torch: 0.6s, import gradio: 0.5s, import ldm: 0.6s, other imports: 0.3s, list SD models: 0.2s, load scripts: 1.2s, load upscalers: 0.1s, create ui: 0.3s).
Creating model from config: /home/bob/code/repos/stable-diffusion/stable-diffusion-webui/configs/v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Loading VAE weights specified in settings: /home/bob/code/repos/stable-diffusion/stable-diffusion-webui/models/Stable-diffusion/nai.vae.pt
Applying optimization: xformers... done.
Textual inversion embeddings loaded(9): bad-artist, bad-artist-anime, bad_prompt, bad_prompt_version2, EasyNegative, emb-avril, fantasycharacter, ng_deepnegative_v1_75t, verybadimagenegative_v1.3
Model loaded in 2.1s (load weights from disk: 0.5s, create model: 0.3s, apply weights to model: 0.5s, load VAE: 0.3s, move model to device: 0.4s).
                                  Loading weights [ed526812bb] from /home/bob/code/repos/stable-diffusion/stable-diffusion-webui/models/Stable-diffusion/Based65-Final-Mix.safetensors
Loading VAE weights specified in settings: /home/bob/code/repos/stable-diffusion/stable-diffusion-webui/models/Stable-diffusion/nai.vae.pt
Applying optimization: xformers... done.
Weights loaded in 1.7s (load weights from disk: 0.2s, apply weights to model: 0.8s, load VAE: 0.3s, move model to device: 0.4s).
INFO: Batch-Checkpoint-Prompt: checkpoint: 1/90
Loading weights [41538124cc] from /home/bob/code/repos/stable-diffusion/stable-diffusion-webui/models/Stable-diffusion/526mixV145_v145.safetensors
Loading VAE weights specified in settings: /home/bob/code/repos/stable-diffusion/stable-diffusion-webui/models/Stable-diffusion/nai.vae.pt
Applying optimization: xformers... done.
Weights loaded in 1.7s (load weights from disk: 0.2s, apply weights to model: 0.8s, load VAE: 0.2s, move model to device: 0.4s).
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 77/77 [00:05<00:00, 13.90it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 77/77 [00:42<00:00,  1.80it/s]
Error completing request                                                                    | 154/13860 [01:02<2:04:52,  1.83it/s]
Arguments: ('task(i3u5s2lxotd2qrn)', 'masterpiece, best quality, woman, realistic, tits, clothes, earrings, rings, necklace, jewelry, on the beach, dragons, waves, sunny day <lora:Astrid-1.5-v1:0.6>, astrid, axe, furs, armor', 'FastNegativeV2, lowres, bad anatomy, bad hands, text, error, distorted face, disfigured body, missing nose, unsymmetrical face, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality, normal quality, jpeg artifacts, signature, watermark, username, blurry, artist name', [], 77, 0, False, False, 1, 1, 7, 2373277865.0, -1.0, 0, 0, 0, False, 512, 512, True, 0.7, 2, 'Latent', 0, 1088, 1088, 0, '', '', [], 4, False, {'ad_model': 'face_yolov8n.pt', 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'inpaint_global_harmonious', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1}, {'ad_model': 'None', 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'inpaint_global_harmonious', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1}, False, False, 'positive', 'comma', 0, False, False, '', 1, '', [], 0, '', [], 0, '', [], True, False, False, False, 0, '526mixV145_v145.safetensors [41538124cc] @index:0,\nAbyssOrangeMix2_Hard.safetensors [cadf2c6654] @index:1,\nAbyssOrangeMix2_nsfw_pruned_fp16_with_VAE.safetensors [e892703c61] @index:2,\nAbyssOrangeMix2SFW_AbyssOrangeMix2Sfw.safetensors [b644d850c9] @index:3,\nAbyssOrangeMix3AOM3_aom3a3.safetensors [eb4099ba9c] @index:4,\naiomonstergirls_v3.safetensors [9ee2133245] @index:5,\naitia_InstantCafe_F1.4_nsfw.safetensors [30929c3f6f] @index:6,\namIReal_V3.safetensors [5abd718d60] @index:7,\nanyhentai_20.safetensors [61bc7001e8] @index:8,\nanyloraCheckpoint_bakedvaeFtmseFp16NOT.safetensors [8a952cafe9] @index:9,\nanything-v4.5-pruned.ckpt [e4b17ce185] @index:10,\nanythingfurry_1.safetensors [3edd27788a] @index:11,\nanythingV5Anything_anythingV5PrtRE.safetensors [7f96a1a9ca] @index:12,\nAOM3A1B_orangemixs.safetensors [5493a0ec49] @index:13,\nbadassMix_lucidDream.safetensors [9294a93172] @index:14,\nBased64mix-V3.safetensors [fdf0096972] @index:15,\nBased65-Final-Mix.safetensors [ed526812bb] @index:16,\ncartoonish_v1.safetensors [07f029f6d1] @index:17,\ncartoonStyle_v02.safetensors @index:18,\ncartunafied_v3.safetensors [894530779c] @index:19,\nCheckpointYesmix_v16Original.safetensors [3e9211917c] @index:20,\ncitrinedreammix_v2BakedVAE.safetensors @index:21,\ncityedge2dtoonmix_v306.safetensors [71b2603dcd] @index:22,\ncoffeemix_v20.safetensors [e2f6e0733a] @index:23,\ncomicBabes_v1.safetensors [9211801e30] @index:24,\ncorneos7thHeavenMix_v2.safetensors [d1e6b263b6] @index:25,\nCounterfeitV30_v30.safetensors [cbfba64e66] @index:26,\ncrosskemonoFurryModel_crosskemono25.safetensors [167e5bcb37] @index:27,\nDegenerate_deliberateV1.safetensors [4eb333dceb] @index:28,\ndeliberate_v11.safetensors [d8691b4d16] @index:29,\ndeliberate_v2.safetensors [9aba26abdf] @index:30,\ndisneyPixarCartoon_v10.safetensors [732d0dd2cf] @index:31,\ndreamscapesDragonfireNEWV20_dsDv20.safetensors [c15090d329] @index:32,\ndreamshaper_6BakedVae.safetensors [c249d7853b] @index:33,\nduchaitenaiart_V453.safetensors [ce4d987a0e] @index:34,\nendlessreality_v2.safetensors [3576ae9275] @index:35,\nexpmixLine_v3.safetensors [ab02baa239] @index:36,\nfantasticmix25D_v20.safetensors [dbbad6ae08] @index:37,\nfantasticmix_v65.safetensors [d4871ddc94] @index:38,\nFluffyCrossKemono.safetensors [c50ed406d5] @index:39,\nfluffyrock-768-lion-low-lr-e38-offset-noise-e6.safetensors [9432e6f806] @index:40,\nfurry_epoch4.ckpt [4160c57f98] @index:41,\nfurrystaber_v40AllVersions.safetensors [67e6e7cec5] @index:42,\nfurryVixens_v20.safetensors [fec0e39345] @index:43,\nfutapocalypse10_v10.safetensors [9887e24890] @index:44,\ngrapefruitHentaiModel_grapefruitv41.safetensors [c590550ea5] @index:45,\nhassakuHentaiModel_v12.safetensors [d3cd6ac55a] @index:46,\nhyper_breasts_1.2k_xl-lr1e-6-10ks-fp16-v3.ckpt [084eaf052e] @index:48,\nkotosmix_v10.safetensors [0f789fc3c4] @index:49,\nlawlassYiffymixFurry_lawlasmixFp16Withvae.safetensors [8f54282b36] @index:50,\nmajicmixRealistic_v5.safetensors [33c9f6dfcb] @index:51,\nmdjrny-v4.safetensors [aba96b389d] @index:52,\nmeinamix_meinaV10.safetensors @index:53,\nmerginmergesofmerge_v25.ckpt [2184ff07cd] @index:54,\nmistoonAmethyst_v20.safetensors [fc896c44f1] @index:55,\nmistoonAnime_v10.safetensors [a49140c6c5] @index:56,\nmistoonEmerald_v20.safetensors [e51d00b968] @index:57,\nmistoonRuby_v20.safetensors [6704d37b6b] @index:58,\nmistoonSapphire_v10.safetensors [b3ef8af57b] @index:59,\nmixProV45Colorbox_v45.safetensors [19dbfda152] @index:60,\nmodel.safetensors [40a255878a] @index:61,\nmoistmix_v2.safetensors [6514f58b5e] @index:62,\nnai.ckpt [89d59c3dde] @index:63,\nneverendingDreamNED_v122BakedVae.safetensors [ecefb796ff] @index:64,\nperfectWorld_v4Baked.safetensors [24a393500f] @index:65,\npixhell_v20.safetensors [333825e530] @index:66,\npony_v4.safetensors [f9ad6f8833] @index:67,\npurefurry_v02.safetensors @index:68,\nqgo10b_qgo10b.safetensors [f8973e771f] @index:69,\nrealdosmix_.safetensors [0d27c62ffa] @index:70,\nrevAnimated_v122.safetensors [4199bcdd14] @index:71,\nsexyToonsFeatPipa_v10.safetensors [bc2d163e9e] @index:72,\nSF_EB_1.0_pruned_vae/SF_EB_1.0_pruned_vae.ckpt [c50ab94df3] @index:73,\nsunshinemix_sunlightmixPruned.safetensors @index:74,\ntfmAmericanCartoons_-inpainting.safetensors [d983f72120] @index:75,\ntfmAmericanCartoons_ckpt.ckpt [5cb7b181ea] @index:76,\ntheAllysMixIII_v10.safetensors [f427dd7a4f] @index:77,\ntheAllysMixIV_v10.safetensors [7e49125818] @index:78,\ntoonify_v10.safetensors [e2a342af33] @index:79,\nuberRealisticPornMerge_urpmv13_pruned.safetensors [40f9701da0] @index:80,\nv1-5-pruned-emaonly.safetensors [6ce0161689] @index:81,\nvae-ft-mse-840000-ema-pruned.safetensors [735e4c3a44] @index:82,\nWestmagic_v10.safetensors [a68f44240f] @index:83,\nwrenchmix_v10.safetensors @index:84,\nxxmix9realistic_v30.safetensors @index:85,\nyiffy-e18.ckpt [64e242ae67] @index:86,\nYuzuGinger_v1-pruned.safetensors [3a76330929] @index:87,\nYuzuLemonChaiLatte_v3.5-pruned.safetensors [ed11f5cd10] @index:88,\nYuzuLemonMilk_v1.5-pruned.safetensors [47caee1481] @index:89,\n3moonNIReal_3moonNIRealV2.safetensors [feadfe3cfe] @index:90,', '{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};\n{prompt};', 0) {}
Traceback (most recent call last):
  File "/home/bob/code/repos/stable-diffusion/stable-diffusion-webui/modules/call_queue.py", line 57, in f
    res = list(func(*args, **kwargs))
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/bob/code/repos/stable-diffusion/stable-diffusion-webui/modules/call_queue.py", line 37, in f
    res = func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^
  File "/home/bob/code/repos/stable-diffusion/stable-diffusion-webui/modules/txt2img.py", line 54, in txt2img
    processed = modules.scripts.scripts_txt2img.run(p, *args)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bob/code/repos/stable-diffusion/stable-diffusion-webui/modules/scripts.py", line 441, in run
    processed = script.run(p, *script_args)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bob/code/repos/stable-diffusion/stable-diffusion-webui/extensions/SD-WebUI-BatchCheckpointPrompt/scripts/batchCheckpointsPrompt.py", line 258, in run
    all_infotexts = self.generate_infotexts(processed_sd_object, all_infotexts, self.get_n_iter(generation_data))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bob/code/repos/stable-diffusion/stable-diffusion-webui/extensions/SD-WebUI-BatchCheckpointPrompt/scripts/batchCheckpointsPrompt.py", line 193, in generate_infotexts
    all_infotexts.append(do_stuff())
                         ^^^^^^^^^^
  File "/home/bob/code/repos/stable-diffusion/stable-diffusion-webui/extensions/SD-WebUI-BatchCheckpointPrompt/scripts/batchCheckpointsPrompt.py", line 186, in do_stuff
    return processing.create_infotext(pc, pc.all_prompts, pc.all_seeds, pc.all_subseeds, position_in_batch=i)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bob/code/repos/stable-diffusion/stable-diffusion-webui/modules/processing.py", line 555, in create_infotext
    uses_ensd = sd_samplers_common.is_sampler_using_eta_noise_seed_delta(p)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bob/code/repos/stable-diffusion/stable-diffusion-webui/modules/sd_samplers_common.py", line 72, in is_sampler_using_eta_noise_seed_delta
    if eta is None and p.sampler is not None:
                       ^^^^^^^^^
AttributeError: 'Processed' object has no attribute 'sampler'
```

I am running Ubuntu Linux 23.04 : 
```
Kernel Version: 6.2.0-23-generic (64-bit) 
Graphics Platform: X11 
Processors: 24 × AMD Ryzen 9 5900X 12-Core Processor 
Memory: 62.7 GiB of RAM 
Graphics Processor: NVIDIA GeForce RTX 4090/PCIe/SSE2 
Manufacturer: Gigabyte Technology Co., Ltd. 
Product Name: X570 AORUS ELITE WIFI 
```

**Comment by h43lb1t0** [comment]
i can't reproduce the error. do you have any additional information that could help me?

**Comment by cdobrich** [comment]
Okay I did some more testing. I ran tests against older versions of the batchCheckpointsPrompt.py. 

This git version properly functions for me - commit dbe53c7dec5acaef253e99f78533bd60c42771ee, Date:   Tue May 16 22:56:43 2023 +0200.

After that, everything else breaks. Does that help?

**Comment by ResleyZ** [comment]
I was getting the same error, and seeing how mine ended with the notation of the ETA noise, which I have set to 31337, I tried again but this time with the ETA Noise Seed Delta set to 0. With ETA noise set to 0, it works fine, I don't get the same sampler error upon starting the script again.


`    Traceback (most recent call last):
      File "C:\AI\Stable Diffusion\Automatic1111\stable-diffusion-webui\modules\call_queue.py", line 55, in f
        res = list(func(*args, **kwargs))
      File "C:\AI\Stable Diffusion\Automatic1111\stable-diffusion-webui\modules\call_queue.py", line 35, in f
        res = func(*args, **kwargs)
      File "C:\AI\Stable Diffusion\Automatic1111\stable-diffusion-webui\modules\txt2img.py", line 54, in txt2img
        processed = modules.scripts.scripts_txt2img.run(p, *args)
      File "C:\AI\Stable Diffusion\Automatic1111\stable-diffusion-webui\modules\scripts.py", line 456, in run
        processed = script.run(p, *script_args)
      File "C:\AI\Stable Diffusion\Automatic1111\stable-diffusion-webui\extensions\SD-WebUI-BatchCheckpointPrompt\scripts\batchCheckpointsPrompt.py", line 248, in run
        self.text_margin_left_and_right = 16
      File "C:\AI\Stable Diffusion\Automatic1111\stable-diffusion-webui\extensions\SD-WebUI-BatchCheckpointPrompt\scripts\batchCheckpointsPrompt.py", line 198, in generate_infotexts
        if name in data:
      File "C:\AI\Stable Diffusion\Automatic1111\stable-diffusion-webui\extensions\SD-WebUI-BatchCheckpointPrompt\scripts\batchCheckpointsPrompt.py", line 191, in do_stuff
        data = {}
      File "C:\AI\Stable Diffusion\Automatic1111\stable-diffusion-webui\modules\processing.py", line 562, in create_infotext
        uses_ensd = sd_samplers_common.is_sampler_using_eta_noise_seed_delta(p)
      File "C:\AI\Stable Diffusion\Automatic1111\stable-diffusion-webui\modules\sd_samplers_common.py", line 72, in is_sampler_using_eta_noise_seed_delta
        if eta is None and p.sampler is not None:
    AttributeError: 'Processed' object has no attribute 'sampler'`

**Comment by h43lb1t0** [comment]
Okay, interesting. I'm looking into this.

**Comment by medledan** [comment]
I am getting the same error. Windows 11.

INFO: Batch-Checkpoint-Prompt: checkpoint: 1/4
100%|██████████████████████████████████████████████████████████████████████████████████| 50/50 [00:04<00:00, 11.20it/s]
Error completing request█████████████████████████████▏                                | 99/200 [02:56<00:09, 10.68it/s]
Arguments: ('task(lfs4xnni6cv25e1)', 'I dont know what to do about my mom\n', 'negative_hand-neg, easynegative:0.5, lowres, low quality:1.4, worst quality:1.4, bad eyes, monochrome, greyscale, signature, watermark, deformed pupils, deformed iris, nude, NSFW, naked', [], 50, 0, False, False, 1, 1, 7, -1.0, -1.0, 0, 0, 0, False, 512, 512, False, 0.35, 2, 'R-ESRGAN 4x+', 15, 0, 0, 0, '', '', [], 5, 0, 0, 4, 512, 512, True, 'None', 'None', 0, True, 'keyword, prompt', 'Mature Adult', 'None', 'model keyword first', 'None', '1.2', 'None', False, 'MultiDiffusion', False, True, 1024, 1024, 96, 96, 48, 1, 'None', 2, False, 10, 1, 1, 64, False, False, False, False, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 1536, 96, True, True, True, False, True, False, 1, False, False, False, 1.1, 1.5, 100, 0.7, False, False, True, False, False, 0, 'Gustavosta/MagicPrompt-Stable-Diffusion', '', <scripts.controlnet_ui.controlnet_ui_group.UiControlNetUnit object at 0x000002DC91AD37C0>, False, False, 'positive', 'comma', 0, False, False, '', 1, '', [], 0, '', [], 0, '', [], True, False, False, False, 0, 5, 'all', 'all', 'all', '', '', '', '1', 'none', False, '', '', 'comma', '', True, '', 'HassanBlend1.5.1.2.safetensors [82089cc267] @index:0,\nttpTheTentacle_v2FP32.safetensors @index:1,\n3dThickCoated_3dThickCoatedV4.ckpt [b3854b01ed] @index:2,\naihMegamerge_anythingInkpunk.safetensors [5e13203e5c] @index:3\n', '{prompt} @index:0;\n\n{prompt} @index:1;\n\n{prompt} @index:2;\n\n{prompt} @index:3;\n\n', 0, None, None, False, 50) {}
Traceback (most recent call last):
  File "G:\Automatic1111\modules\call_queue.py", line 57, in f
    res = list(func(*args, **kwargs))
  File "G:\Automatic1111\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "G:\Automatic1111\modules\txt2img.py", line 54, in txt2img
    processed = modules.scripts.scripts_txt2img.run(p, *args)
  File "G:\Automatic1111\modules\scripts.py", line 441, in run
    processed = script.run(p, *script_args)
  File "G:\Automatic1111\extensions\SD-WebUI-BatchCheckpointPrompt\scripts\batchCheckpointsPrompt.py", line 248, in run
    all_infotexts = self.generate_infotexts(processed_sd_object, all_infotexts, all_batchParams[i].batch_count)
  File "G:\Automatic1111\extensions\SD-WebUI-BatchCheckpointPrompt\scripts\batchCheckpointsPrompt.py", line 198, in generate_infotexts
    all_infotexts.append(do_stuff())
  File "G:\Automatic1111\extensions\SD-WebUI-BatchCheckpointPrompt\scripts\batchCheckpointsPrompt.py", line 191, in do_stuff
    return processing.create_infotext(pc, pc.all_prompts, pc.all_seeds, pc.all_subseeds, position_in_batch=i)
  File "G:\Automatic1111\modules\processing.py", line 555, in create_infotext
    uses_ensd = sd_samplers_common.is_sampler_using_eta_noise_seed_delta(p)
  File "G:\Automatic1111\modules\sd_samplers_common.py", line 72, in is_sampler_using_eta_noise_seed_delta
    if eta is None and p.sampler is not None:
AttributeError: 'Processed' object has no attribute 'sampler'

**Comment by h43lb1t0** [comment]
seems to be a problem with custom eta noise settings. I'll try to find out how to fix it

**Comment by jb-lopez** [comment]
I'm getting the same error on Debian 11. I tried unsetting ENSD (empty textbox) and I still got the same error. But I can confirm that setting it to 0 does work.

**Comment by h43lb1t0** [comment]
fixed in commit [ed335e1](https://github.com/h43lb1t0/SD-WebUI-BatchCheckpointPrompt/commit/ed335e1dbe7eba6f1fd12388b4275b745a79dddc)
